crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -302,19 +363,14 @@ impl IndexerStore for PgIndexerStore {
                 if reverse {
                     builder.filter(move_calls_dsl::id.le(start_sequence))
                         .order(move_calls_dsl::id.desc())
-                        // id is needed in the select clause to make distinct work
-                        .select((move_calls_dsl::transaction_digest, move_calls_dsl::id))
-                        .distinct()
```

Comment:
I think `distinct()` does not work here not because the `DISTINCT ON` problem that you mentioned above, which is different from `distinct`(https://stackoverflow.com/questions/50846722/what-is-the-difference-between-postgres-distinct-vs-distinct-on).

It's actually expected that the existing usage of `distinct()` fail to return distinct digests because the `id` field is always unique.

See example usage here: https://docs.diesel.rs/master/diesel/prelude/trait.QueryDsl.html#example let me know if it does not work

Reply:
Yeah that's what I meant:
the existing distinct() will return values but wrong ones, instead to get correct distinct, we need `distinct_on(txn_digesst)`, then the problem mentioned in another comment showed up.

Reply:
Synced offline, for posterity,`id` is required here because we are ordering by `id` and hence it must be selected